File: 125-valid-palindrome/125-valid-palindrome.cpp
class Solution {
public:
    bool isAlphanumeric(char c)
    {
        if((c>='0'&&c<='9')||((c>=97)&&(c<=122)))
           return true;
        return false;
    }
    bool isPalindrome(string s) {
        transform(s.begin(),s.end(),s.begin(),::tolower);
        cout<<s;
        int i,j;
        i=0;
        j=s.length()-1;
        while(i<j)
        {
            while(!isAlphanumeric(s.at(i)))
            {
                cout<<s.at(i);
                ++i;
                if(i==j)
                    return true;
            }
            while(!isAlphanumeric(s.at(j)))
            {
                cout<<s.at(j);
                --j;
                if(i==j)
                    return true;
            }
            if(s.at(i)!=s.at(j))
                return false;
            --j;
            ++i;
        }
        return true;
    }
};

File: 125-valid-palindrome/125-valid-palindrome.java
class Solution {
    public boolean isPalindrome(String s) {
       String res = s.toLowerCase();
        int left = 0;
        //res.toCharArray();
        int right = res.length()-1;
        while(left < right){
            if(res.charAt(left) < '0' || (res.charAt(left) > '9' && res.charAt(left)<'a') || res.charAt(left)>'z'){
                left++;
                continue;
            }else if(res.charAt(right) < '0' || (res.charAt(right) > '9' && res.charAt(right)<'a') || res.charAt(right)>'z'){
                right--;
                continue;
            }
            if(res.charAt(left) != res.charAt(right)){
                return false;
            }
            left++;
            right--;
        }
        return true;
    }
}